examples/distributed/pyg/README.md

```diff
@@ -0,0 +1,97 @@
+
+This example shows how to use distributed PyG to train PyG models in a distributed scenario with multiple CPU nodes.
+
+### Step 1: Prepare and partition the data
+
+In distributed training, each node in the cluster holds a partition of the graph. Before the training starts, we partition the dataset into multiple partitions, each of which corresponds to a specific training node.
+
+Here, we use `ogbn-products` and partition it into two partitions (in default) by the [partition example](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/distributed/pyg/partition_graph.py) :
+
+```bash
+python partition_graph.py --dataset=ogbn-products --root_dir=../../../data --num_partitions=2
+```
+
+**Caution:** Partitioning with METIS is non-deterministic!
+All nodes should be able to access the same partition data, therefore generate partition on one node and copy the data to all members of the cluster, or place the folder in shared location.
+
+The generated partition will have the folder below.
+
+<img width="350" alt="partition-graph" src="https://github.com/pyg-team/pytorch_geometric/assets/58218729/2169e362-0259-4ac4-ab5e-8500b6b5bf4a">
+
+### Step 2: Run the example in each training node
+
+To run the example you can execute the commands in each node or use the provided launch script.
+
+## Option A: Manual execution
+
+You should change the `master_addr` to the IP of `node#0`. Make sure that correct `node_rank` is provided, with master node assigned rank 0. The `dataset_root_dir` should point to the head directory where your partition is placed, i.e. `../../data/partitions/ogbn-products/2-parts`.
+
+```bash
+# Node 0:
+python distributed_cpu.py \
+  --dataset=ogbn-products
+  --dataset_root_dir=<partition folder directory> \
+  --num_nodes=2 \
+  --node_rank=0 \
+  --master_addr=<master ip>
+
+# Node 1:
+python distributed_cpu.py \
+  --dataset=ogbn-products
+  --dataset_root_dir=<partition folder directory> \
+  --num_nodes=2 \
+  --node_rank=1 \
+  --master_addr=<master ip>
+```
+
+## Option B: Launch script
+
+See PR [#8241](https://github.com/pyg-team/pytorch_geometric/pull/8241)
+
+There are two methods to run the distributed e2e test with one script in one terminal for multiple nodes.
+
+1. launch.py
+
+```bash
+    python launch.py
+          --workspace {workspace}/pytorch_geometric
+          --num_nodes 2
+          --dataset_root_dir {dataset_dir}/mag/2-parts
+          --dataset ogbn-mag
+          --batch_size 1024
+          --learning_rate 0.0004
+          --part_config {dataset_dir}/mag/2-parts/ogbn-mag-partitions/META.json
+          --ip_config {workspace}/pytorch_geometric/ip_config.yaml
+          'cd /home/user_xxx; source {conda_envs}/bin/activate; cd {workspace}/pytorch_geometric; {conda_envs}/bin/python
+          {workspace}/pytorch_geometric/examples/pyg/distributed_cpu.py --dataset=ogbn-mag --logging --progress_bar --ddp_port=12345'
```

Comment:
Please make ddp_port value consistent with the previous one declared in this document.